File: Graphs/insert_vertex.c
/* File Name: insert_vertex.c */

#include <stdio.h>
#include <stdlib.h>

#define MAX 100

int graph[MAX][MAX];   // Adjacency matrix
int vertices = 0;

void insertVertex() {
    vertices++;
    printf("Vertex %d inserted successfully.\n", vertices - 1);
}

void displayGraph() {
    printf("\nAdjacency Matrix:\n");
    for(int i = 0; i < vertices; i++) {
        for(int j = 0; j < vertices; j++) {
            printf("%d ", graph[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int choice;

    while(1) {
        printf("\n1. Insert Vertex\n2. Display Graph\n3. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch(choice) {
            case 1:
                insertVertex();
                break;
            case 2:
                displayGraph();
                break;
            case 3:
                exit(0);
            default:
                printf("Invalid choice\n");
        }
    }
    return 0;
}

File: Graphs/insert_vertex.py
# File Name: insert_vertex.py

graph = []

def insert_vertex():
    graph.append([])
    print(f"Vertex {len(graph) - 1} inserted successfully.")

def display_graph():
    print("\nAdjacency List:")
    for i in range(len(graph)):
        print(f"{i} -> {graph[i]}")

while True:
    print("\n1. Insert Vertex")
    print("2. Display Graph")
    print("3. Exit")
    choice = int(input("Enter your choice: "))

    if choice == 1:
        insert_vertex()
    elif choice == 2:
        display_graph()
    elif choice == 3:
        break
    else:
        print("Invalid choice")
